### ISSUE TITLE
Docker container for nodes?

### ISSUE THREAD

**Issue by loth** [weizhouapache/docker-cloudstack-management#1]
Hello,

Has any work been done on a node image?

Thanks.

**Comment by weizhouapache** [comment]
@loth 
do you mean the docker image for `cloudstack-agent` service ?

I wonder how the `cloudstack-agent` process in docker container perform some command on the kvm  host, for example virsh, qemu, some scripts, shell commands.

**Comment by loth** [comment]
Yeah for the agent service, basically so we can dockerize a full hypervisor

Openstack does this in this project https://github.com/openstack/kolla maybe worth a look?

**Comment by weizhouapache** [comment]
> Yeah for the agent service, basically so we can dockerize a full hypervisor
> 
> Openstack does this in this project https://github.com/openstack/kolla maybe worth a look?

@loth 
interesting

as I understand, some services like cloudstack-agent, libvirt (and other dependencies) are running in the container, while vms are running on the hypervisor, is it right  ?

**Comment by loth** [comment]
Yes i believe they expose the qemu-kvm device to a special container which houses the VMs

**Comment by weizhouapache** [comment]
I see. you want a docker container to mock a kvm host and create vms inside it, right ?
for test purpose, it should be feasible.  I think nobody will use it in production :smile:

**Comment by weizhouapache** [comment]
Maybe I am wrong

cloudstack agent looks like a combinaton of nova, neutron, it also manages volumes ( by libvirt or access storage directly).

It looks kolla can create containers for libvirt and openvswitch. I am able to manage vms on a kvm host using libvirt.
However, for neutron and storage, I do not find a doc saying they can use containers . 
https://docs.openstack.org/kolla-ansible/latest/reference/index.html

If we run cloudstack-agent in a container, it is difficult to manage the network and storage configurations.
